### PR TITLE
Fix block-scoped doctest tags with `\` line continuations

### PR DESCRIPTION
**Pull request by mkoeppe**
Fixes https://github.com/sagemath/sage/pull/36033#issuecomment-1665062002

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

- Depends on #36025 (merged here)

### CODE REVIEW

**Comment by kwankyu** [comment]
```diff
--- a/src/sage/dynamics/arithmetic_dynamics/wehlerK3.py
+++ b/src/sage/dynamics/arithmetic_dynamics/wehlerK3.py
@@ -474,7 +474,7 @@ class WehlerK3Surface_ring(AlgebraicScheme_subscheme_product_projective):
            sage: Z = x0^2*y0^2 + 3*x0*x1*y0^2 + x1^2*y0^2 + 4*x0^2*y0*y1 + 3*x0*x1*y0*y1 \
            - 2*x2^2*y0*y1 - x0^2*y1^2 + 2*x1^2*y1^2 - x0*x2*y1^2 - 4*x1*x2*y1^2 \
            + 5*x0*x2*y0*y2 \
-           - 4*x1*x2*y0*y2 + 7*x0^2*y1*y2 + 4*x1^2*y1*y2 + x0*x1*y2^2 + 3*x2^2*y2^2 \
+           - 4*x1*x2*y0*y2 + 7*x0^2*y1*y2 + 4*x1^2*y1*y2 + x0*x1*y2^2 + 3*x2^2*y2^2
            sage: Y = x0*y0 + x1*y1 + x2*y2
            sage: X = WehlerK3Surface([Z, Y])
            sage: T = PP(1, 1, 0, 1, 0, 0)
```

File: src/sage/coding/code_constructions.py [hunk 1]
```diff
@@ -732,8 +732,7 @@ def ToricCode(P,F):
          sage: C.minimum_distance()
          24
          sage: C.minimum_distance(algorithm="guava")  # optional - gap_packages (Guava package)
-         ...
-         24
+         ...24
          sage: C = codes.ToricCode([[-2,-2],[-1,-2],[-1,-1],[-1,0],
```
Comment (kwankyu) [inline src/sage/coding/code_constructions.py:736]: Why `...`?

I get just 
```
sage:  C.minimum_distance(algorithm="guava")  
24
```
after installing guava.

Comment (mkoeppe) [reply src/sage/coding/code_constructions.py:736]: I would guess it accounts for sporadic output from GAP

**Comment by kwankyu** [comment]
We missed this one:
```diff
--- a/src/sage/tests/book_stein_ent.py
+++ b/src/sage/tests/book_stein_ent.py
@@ -21,10 +21,10 @@ sage: factor(2007)
 sage: factor(31415926535898)
 2 * 3 * 53 * 73 * 2531 * 534697
 sage: n = 7403756347956171282804679609742957314259318888\
-...9231289084936232638972765034028266276891996419625117\
-...8439958943305021275853701189680982867331732731089309\
-...0055250511687706329907239638078671008609696253793465\
-...0563796359
+....: 9231289084936232638972765034028266276891996419625117\
+....: 8439958943305021275853701189680982867331732731089309\
+....: 0055250511687706329907239638078671008609696253793465\
+....: 0563796359
 sage: len(n.str(2))
 704
 sage: len(n.str(10))
```

**Comment by mkoeppe** [comment]
Thanks!

**Review by kwankyu — APPROVED** [review]
Thanks.

**Comment by vbraun** [comment]
Merge conflict

**Comment by mkoeppe** [comment]
Merged with #36025 to resolve merge conflict.

**Pull request merged**